train.py

```diff
@@ -0,0 +1,122 @@
+import os
+import time
+
+import hydra
+import torch
+import psutil
+import imageio
+import numpy as np
+from omegaconf import DictConfig
+from torchvision.utils import save_image
+
+import wandb
+from utils import get_interpolations
+from data_loader import get_data_loaders
+from models.autoencoder import Autoencoder
+from train_utils import test_epoch, train_epoch, prepare_loss_function
+
+
+def initialize_model(cfg: DictConfig, train_loader, test_loader, input_size):
+    if cfg.model.type == "AE":
+        model = Autoencoder(cfg, train_loader=train_loader, test_loader=test_loader, input_size=input_size)
+    else:
+        raise NotImplementedError("Model type not implemented")
+    return model
+
+
+def train(cfg: DictConfig, autoencoder: Autoencoder, loss_function: torch.nn.Module):
+    os.makedirs(cfg.logger.results_path, exist_ok=True)
+    os.makedirs(cfg.logger.checkpoint_path, exist_ok=True)
+
+    torch.manual_seed(cfg.system.seed)
+
+    train_autoencoder(cfg, autoencoder, loss_function)
+    draw_interpolation_grid(cfg, autoencoder)
+
+
+def train_autoencoder(cfg: DictConfig, autoencoder: Autoencoder, loss_function: torch.nn.Module):
+    total_start_time = time.time()
+    process = psutil.Process()
+
+    optimizer = torch.optim.Adam(autoencoder.model.parameters(), lr=cfg.train.lr)
+    for epoch in range(1, cfg.train.epochs + 1):
+        train_epoch(
+            autoencoder, autoencoder.train_loader, optimizer, autoencoder.device, cfg.train.log_interval, epoch, loss_function
+        )
+        # test_epoch(autoenc, autoenc.test_loader, autoenc.device)
+        test_loss = test_epoch(autoencoder, autoencoder.test_loader, autoencoder.device, loss_function)
+
+        # save checkpoint
+        checkpoint = {
+            "epoch": epoch,
+            "model_state_dict": autoencoder.model.state_dict(),
+            "optimizer_state_dict": optimizer.state_dict(),
+            "config": cfg,  # Saving the config used for this training run
+        }
+        checkpoint_path = "{}/{}_{}_checkpoint_epoch_{}.pt".format(
+            cfg.logger.checkpoint_path, cfg.model.type, cfg.dataset.name, epoch
+        )
+        torch.save(checkpoint, checkpoint_path)
+
+        memory_usage = process.memory_info().rss / (1024 * 1024)  # in megabytes
+
+        # Pseudocode for wandb logging
+        if cfg.logger.enable_wandb:
+            wandb.log({"epoch": epoch, "test_loss": test_loss, "memory_usage": memory_usage})
```

Comment:
```suggestion
            wandb.log({"epoch": epoch, "test/loss": test_loss, "memory_usage": memory_usage})
```

Using `test/METRIC_NAME` format will create a separate section in wandb. Also, you should include the loss value for both train and test split whenever possible.

Reply:
Updated, logs the histogram of weights of the first convolution as well.